veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,258 @@
+- Enhanced IPv6 support.
+- High network throughput, on par with Masquerade and Bridge bindings.
+
+## Non Goals
+
+- Supporting network protocols other those that are [currently supported](https://passt.top/passt/about/#protocols).
+- Supporting CNIs that do not provide sticky IP addresses during migration (migration succeeds, but existing connections break).
+
+## Definition of Users
+
+- Cluster Admins
+- VM/Namespace owners
+- Guest VM users – typically unaware of their network binding
+
+## User Stories
+
+- As a Guest VM user, I want optimal network performance comparable to bridge and masquerade, ensuring my virtualized applications perform well.  
+- As a Cluster Admin and as a Namespace Owner, I want observability and service meshes for enhanced monitoring, encryption, and load balancing.
+- As a Cluster Admin and as a Namespace Owner, I require seamless VM migration to perform infrastructure upgrades without affecting business continuity, preserving connectivity event for sensitive applications and protocols.
+- As a Cluster Admin, I want to make sure that I can upgrade existing VMs to the new binding and that the procedure to do so is clear.
+
+## Repos
+
+- [https://github.com/kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)
+
+## Design
+### Live Migration with passt
+![Live Migration with passt ](./images/passtMigrationDiagram.png)
+
+### About passt
+
+`passt` is a lightweight user-space networking solution designed for virtualized environments. It allows guest VMs to access the host’s network stack without requiring root privileges or additional network bridges. 
+It is already used in KubeVirt as a network binding plugin, and is documented in the [user guide](https://kubevirt.io/user-guide/network/net_binding_plugins/passt/#passt-binding).
+
+### Support live migration
+
+The [Live Migration](ttps://kubevirt.io/user-guide/compute/live_migration/#live-migration) process triggers a chain of events, which at a high level networking perspective involves the following steps:
+- A target pod is prepared, with applicable network plumbing.
+- Libvirt is executed. It receives the DomainXML data from the source VM and it also launches the `passt` binary, before executing QEMU which starts the VM.
+- Once both source and target VMs are up, depending on [the migration strategy](https://kubevirt.io/user-guide/compute/live_migration/#understanding-different-migration-strategies) a switchover occurs from the source to the target.
+- the passt instance running on the source sets the TCP_REPAIR socket option on TCP sockets, thus freezing the state of the connection and allowing a dump of all the information needed to restore it (sequences, congestion window parameters, pending data queues - IP addresses and ports are already known.).
+- It communicates the state to peer `passt` process running in the target host, via [facilities provided by vhost-user](https://qemu-project.gitlab.io/qemu/interop/vhost-user).html#migrating-back-end-state)
```

Comment:
Over which medium does this communication occurs?

Reply:
Correct, that is, QEMU exports a channel to source and target instances of passt, which assume the role of vhost-user back-ends. But given that https://qemu-project.gitlab.io/qemu/interop/vhost-user.html#migrating-back-end-state is referenced, that already explains the whole story...

Reply:
Thank you for explaining @sbrivio-rh!